### DISCUSSION TITLE
Has anybody written the VSCode plugins/extensions yet?

### DISCUSSION THREAD

**Discussion by DavidPoliakoff** [spack/spack#44144] [Q&A]
Long time no see to DOE (and adjacent) friends. I've turned into a pretty "VSCode is the way and the truth" kind of developer.

Has anybody written the VSCode plugins yet? Specifically

- [ ] As Spack installs packages, I get helpful notices in VSCode
- [ ] Let me drive Spack from VSCode (install packages, see specs)
- [ ] (more than a plugin/extension) make VSCode "feel as if it runs commands in a Spack environment" (if I run "Test", by default I get a correctly-ish configured version of the Spack compiler from my chosen environment, or the root otherwise)

The first two don't feel all that intimidating. Built right (something gRPC-like) it wouldn't even have to be VSCode specific, servers could listen for updates and enqueue requests, so residents of other IDEs could play.

I dug in a bit to Spack's code and found the [CDash/Junit reporters](https://spack.readthedocs.io/en/latest/spack.reporters.html), so you even have (some of) the event emitting infra.

Not making an FR because this is just "David chasing shiny objects," not any kind of urgent need.

**Comment by alalazo** [comment]
> Has anybody written the VSCode plugins yet?

Not that I know of :slightly_smiling_face:

**Comment by DavidPoliakoff** [comment]
Sweet, I'll give it a more thorough go.

<img width="1531" alt="Screenshot 2024-05-17 at 1 32 13 PM" src="https://github.com/spack/spack/assets/4220361/1d898bae-8621-4259-bdce-033d72c53a46">


In early messing around, I have it so that you can have your Spack installs show up in VSCode as alerts, as well as getting more detailed spec data in the treeview on the side. Only Spack change was to add a reporter

Next up is enqueueing requests to install

**Comment by hfbauman** [comment]
@DavidPoliakoff Did you ever get this working? I'm looking for something similar (in particular, the ability to run CTest through VS Code with my spack environment properly configured). I don't find your extension on the marketplace or the source code anywhere...
